sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectSummary.groovy

```diff
@@ -13,79 +13,33 @@ import life.qbic.datamodel.dtos.projectmanagement.Project
  */
 @EqualsAndHashCode
 class ProjectSummary {
-    String code
-    String title
+    private String code
+    private String title
     int samplesReceived
     int samplesQcFailed
-
-    private ProjectSummary(Builder builder) {
-        this.code = builder.code
-        this.title = builder.title
-        this.samplesReceived = builder.samplesReceived
-        this.samplesQcFailed = builder.samplesQcFailed
+    int sampleDataAvailable
+
+    ProjectSummary(Project project, int samplesReceived, int samplesQCFailed, int samplesDataAvailable) {
+        this.code = project.projectId.projectCode.toString()
+        this.title = project.projectTitle
+        this.samplesReceived = samplesReceived
+        this.samplesQcFailed = samplesQCFailed
+        this.sampleDataAvailable = samplesDataAvailable
     }
 
-    static class Builder {
-        private String code
-        private String title
-        private int samplesReceived
-        private int samplesQcFailed
-
-        /**
-         * Constructs a builder and sets the code and title.
-         * @param code the project code to be used to build a ProjectSummary
-         * @param title the project title to be used to build a ProjectSummary
-         * @since 1.0.0
-         */
-        Builder(String code, String title) {
-            this.code = code
-            this.title = title
-            this.samplesReceived = 0
-            this.samplesQcFailed = 0
-        }
-
-        /**
-         * Constructs a builder using information from a project dto
-         * @param projectDto the project dto to be read
-         * @since 1.0.0
-         */
-        Builder(Project projectDto) {
-            this.code = projectDto.projectId.projectCode.toString()
-            this.title = projectDto.projectTitle
-            this.samplesReceived = 0
-            this.samplesQcFailed = 0
-        }
-
-        /**
-         * Sets the number of received samples
-         * @param number
-         * @return the current Builder with the number set
-         * @since 1.0.0
-         */
-        Builder samplesReceived(int number) {
-            this.samplesReceived = number
-            return this
-        }
-
+    ProjectSummary(Project project) {
+        this.code = project.projectId.projectCode.toString()
+        this.title = project.projectTitle
+        this.samplesReceived = 0
+        this.samplesQcFailed = 0
+        this.sampleDataAvailable = 0
+    }
```

Comment:
```suggestion
    ProjectSummary(String code, String title) {
        this.code = code
        this.title = title
        this.samplesReceived = 0
        this.samplesQcFailed = 0
        this.sampleDataAvailable = 0
    }
    
    static ProjectSummary of(Project project) {
        String code = project.projectId.projectCode.toString()
        String title = project.projectTitle
        return new ProjectSummary(code, title)
    }
```
We can change that to a factory which is a nicer way here than a constructor with the Project DTO. This way we only have primitive types in our constructor.

Reply:
If we would need the opposite mapping we could provide an additional method `Project toProject()`.